### ISSUE TITLE
Svg

### ISSUE THREAD

**Issue by gedw99** [franzsilva/flutter_launcher_icons#20]
I have used this before and it's such a Time saver.

Do you think svg support it possible ?

There is a dart package that converts svg to a bitmap on pub.
So you could use that.

I have been using the svg approach for all images in my apps because it ensures it looks good on a tablet or a little mobile. Once a bit at has been generated at runtime I save it and use that from then on. This allows the image to be cached.

Just thought this was worth mentioning.

**Comment by franzsilva** [comment]
hi @gedw99 , Yes it is possible we will look into this and get back to you.

**Comment by gedw99** [comment]
Here it is:

https://github.com/flutter/engine/pull/5060

example
````
[if(feature='Flutter.Image.encodeImage')]
enum EncodedImageFormat {
  [if(feature='Flutter.Image.encodeImage.jpeg')]
  JPEG,
  [if(feature='Flutter.Image.encodeImage.png')]
  PNG,
  [if(feature='Flutter.Image.encodeImage.webp')]
  WEBP,
}

class Image ... {
  [if(feature='Flutter.Image.toByteData')]
  Future<ByteData> toByteData({EncodingFormat format: const EncodingFormat.jpeg()}) { ... }
  
  [if(feature='Flutter.Image.encodeImage')]
  void encodeImage(EncodedImageFormat format, int quality, ImageEncoderCallback callback) { ... }
}
````

also checkout
https://github.com/dnfield/flutter_path_drawing

**Comment by NatoBoram** [comment]
I tried to use this package, but turns out it doesn't support SVG.

```
Android minSdkVersion = 16
Creating default icons Android
Overwriting the default Android launcher icon with a new icon
NoSuchMethodError: The getter 'width' was called on null.
Receiver: null
Tried calling: width
pub finished with exit code 2
```

Ping me when SVG is supported :)

**Comment by connectety** [comment]
I took a sharp look at implmenting SVGs.
The Problem is rendering/converting the SVG to PNG.
I was not able to find a library with working SVG which does not depend on flutter.

I believe that, if it depends on flutter it must be run with the flutter command like an Flutter-app.
I would be pleased if someone tells me I am incorrect and also how to run a CLI-app which depends on dart:ui (flutter exclusive I think).

Probably correct ways to do this:
- a fork of [flutter_svg](https://github.com/dnfield/flutter_svg), which does not have any dependencies on flutter.
- a standalone library just for this


Workarounds:
- running the converting stuff via `flutter test`, which again gets called via console
- Some binaries for Windows, Linux & Mac which can convert to PNG

**Comment by connectety** [comment]
I started implemented SVG support using inkscape.

Before merging the changes I want to:
-  add MacOS support (I do not have a Mac for that :( and would require the help of someone who does)
-  support custom install paths
-  test the SVG support some more
-  get feedback on what to improve

If you want to help / take a look at it  check out my branch: https://github.com/connectety/flutter_launcher_icons/tree/svg

**Comment by MarkOSullivan94** [comment]
@connectety it's work in progress right now, wouldn't recommend anyone to use it until it's merged into master branch

**Comment by mqus** [comment]
I also would really like this feature.
Is there any progress? Is there anything we can do to help?

**Comment by Nioux** [comment]
We are interrested to use it in our project too, but svg support is a requirement right now. Is there a planned date for this feature ?

**Comment by PolymathWhiz** [comment]
I would really like this feature merged

**Comment by masseelch** [comment]
If you are on unix simply use `convert logo.svg logo.png` and you have a valid png to use for this package. I make a bash script which just converts svg to png, then calls this package.

**Comment by N3rv0us** [comment]
I would really like this feature merged, too

**Comment by nohkumado** [comment]
same here, all my artwork for my projects is done in svg....

**Comment by bradintheusa** [comment]
Dart's svg support is not great. Are you able to load them ok?

**Comment by HannesGitH** [comment]
> If you are on unix simply use `convert logo.svg logo.png` and you have a valid png to use for this package. I make a bash script which just converts svg to png, then calls this package.

problem is that then the icons would still be png in the final package which is not desirable, the android_vector xml assets would be much better

**Comment by Sun3** [comment]
Any updates on support svg?

Thanks

**Comment by aastrouski** [comment]
Any updates?

**Comment by HannesGitH** [comment]
I don't think so, but I might be able to find some time for this the coming weeks, I'll open a PR and let you know then (in case it's a relatively easy fix)

**Comment by normidar** [comment]
Any updates?

**Comment by ankitkapoor2493** [comment]
Any update on this feature?

**Comment by dingshaohua-com** [comment]
andriod native development is support too!

**Comment by normidar** [comment]
Maybe we should make a Dart svg tool first.

**Comment by TiagoDanin** [comment]
Any update on this feature?

**Comment by normidar** [comment]
I am making a pure_svg library but it’s not perfect, can anybody help me?（pure svg tool can make this feature possible）

link: https://github.com/normidar/pure_svg

It forked from [flutter_svg](https://github.com/flutter/packages/tree/main/third_party/packages/flutter_svg)
And depend on [pure_ui](https://github.com/normidar/pure_ui)

**Comment by ryansuhartanto** [comment]
Any updates on this? It's been 7 years.
